oss-internship-2020/gdal/raster_to_gtiff/raster_to_gtiff.cc

```diff
@@ -0,0 +1,188 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdlib>
+#include <filesystem>
+#include <iostream>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "sandboxed_api/sandbox2/util/fileops.h"
+
+#include "gdal_sandbox.h"  // NOLINT(build/include)
+#include "get_raster_data.h"  // NOLINT(build/include)
+
+namespace {
+
+#define SAPI_FAIL_IF_NOT(x, y)               \
+  if (!(x)) {                                \
+    return absl::FailedPreconditionError(y); \
+  }
+
+inline constexpr absl::string_view kDriverName = "GTiff";
+inline constexpr absl::string_view kProjDbEnvVariableName = "PROJ_PATH";
+inline constexpr absl::string_view kDefaultProjDbPath 
+    = "/usr/local/share/proj/proj.db";
+
+std::optional<std::string> GetProjDbPath() {
+  const char* proj_db_path_ptr = std::getenv(kProjDbEnvVariableName.data());
+
+  std::string proj_db_path = proj_db_path_ptr == nullptr ? 
+      std::string(kDefaultProjDbPath) : std::string(proj_db_path_ptr);
+
+  if (!std::filesystem::exists(proj_db_path)) {
+    return std::nullopt;
+  }
+
+  return proj_db_path;
+}
+
+absl::Status SaveToGTiff(gdal::sandbox::tests::parser::RasterDataset bands_data, 
+    std::string out_file) {
+  using namespace gdal::sandbox;
+
+  std::string output_data_folder = "";
+
+  if (!sandbox2::file_util::fileops::RemoveLastPathComponent(out_file,
+      &output_data_folder)) {
+    return absl::FailedPreconditionError("Error getting output file directory");
+  }
+
+  std::optional<std::string> proj_db_path = GetProjDbPath();
+  SAPI_FAIL_IF_NOT(proj_db_path != std::nullopt,
+      "Specified proj.db does not exist");
+
+  GdalSapiSandbox sandbox(output_data_folder, std::move(proj_db_path.value()));
+  SAPI_RETURN_IF_ERROR(sandbox.Init());
+  gdalApi api(&sandbox);
+  SAPI_RETURN_IF_ERROR(api.GDALAllRegister());
+
+  sapi::v::ConstCStr driver_name_ptr(kDriverName.data());
```

Comment:
I know `c_str()` does the same since C++11 but it does indicate the intention slightly better.